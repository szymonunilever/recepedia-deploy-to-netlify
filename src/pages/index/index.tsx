import { graphql } from 'gatsby';
import React from 'react';
import Layout from 'src/components/Layout/Layout';
import SEO from 'src/components/Seo/Seo';
import { Text, TagName } from 'src/components/lib/components/Text';

import { RecipeListing } from 'src/components/lib/components/RecipeListing';
import Hero from 'src/components/lib/components/Hero';
import { findPageComponentContent } from 'src/utils';
import { RecipeItem } from 'src/components/lib/components/RecipeListing/partials';

const HomePage = ({ data }: HomePageProps) => {
  const page = data.allPage.edges[0].node;
  const components = page.components.items;
  const recipes = data.allRecipe.nodes;

  return (
    <Layout>
      <SEO title="Recepedia Home" />

      <Text tag={TagName['h1']} text={data.allPage.edges[0].node.title} />
      <section>
        <RecipeListing
          content={findPageComponentContent(
            components,
            'RecipeListing',
            'LatestAndGreatest'
          )}
          list={recipes}
        />
      </section>

      <section>
        <RecipeListing
          content={findPageComponentContent(
            components,
            'RecipeListing',
            'TopRecipes'
          )}
          list={recipes}
        />
      </section>

      <section>
        <Hero
          content={findPageComponentContent(components, 'Hero')}
          viewType="Image"
        />
      </section>
    </Layout>
  );
};

export default HomePage;

export const pageQuery = graphql`
  {
    allPage(filter: { type: { eq: "Home" } }) {
      edges {
        node {
          ...PageFields
        }
      }
    }
    allRecipe(skip: 10) {
      nodes {
        ...RecipeFields
      }
    }
  }
`;

interface HomePageProps {
  data: {
    allPage: {
      edges: Edge<PageNode>[];
    };
    allRecipe: {
      nodes: RecipeItem[];
    };
  };
}

interface Edge<T> {
  node: T;
}

interface PageNode {
  components: {
    items: {
      name: string;
      content: any;
    }[];
  };
  title: string;
  type: string;
}
