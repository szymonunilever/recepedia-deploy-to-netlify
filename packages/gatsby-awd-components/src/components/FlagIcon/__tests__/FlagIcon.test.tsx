import { mount, ReactWrapper } from 'enzyme';
import React from 'react';
import { FlagIcon } from '../index';
import toJson from 'enzyme-to-json';

describe('<FlagIcon />', () => {
  let wrapper: ReactWrapper;
  let props = {
    react: React,
    code: 'km',
  };

  beforeEach(() => {
    wrapper = mount(<FlagIcon {...props} />);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('matches the snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
