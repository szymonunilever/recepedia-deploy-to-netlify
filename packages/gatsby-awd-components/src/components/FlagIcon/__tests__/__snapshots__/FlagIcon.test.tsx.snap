// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FlagIcon /> matches the snapshot 1`] = `
<FlagIcon
  code="km"
  react={
    Object {
      "Children": Object {
        "count": [Function],
        "forEach": [Function],
        "map": [Function],
        "only": [Function],
        "toArray": [Function],
      },
      "Component": [Function],
      "Fragment": Symbol(react.fragment),
      "Profiler": Symbol(react.profiler),
      "PureComponent": [Function],
      "StrictMode": Symbol(react.strict_mode),
      "Suspense": Symbol(react.suspense),
      "__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED": Object {
        "IsSomeRendererActing": Object {
          "current": false,
        },
        "ReactComponentTreeHook": Object {},
        "ReactCurrentBatchConfig": Object {
          "suspense": null,
        },
        "ReactCurrentDispatcher": Object {
          "current": Object {
            "readContext": [Function],
            "useCallback": [Function],
            "useContext": [Function],
            "useDebugValue": [Function],
            "useDeferredValue": [Function],
            "useEffect": [Function],
            "useImperativeHandle": [Function],
            "useLayoutEffect": [Function],
            "useMemo": [Function],
            "useReducer": [Function],
            "useRef": [Function],
            "useResponder": [Function],
            "useState": [Function],
            "useTransition": [Function],
          },
        },
        "ReactCurrentOwner": Object {
          "current": null,
        },
        "ReactDebugCurrentFrame": Object {
          "getCurrentStack": null,
          "getStackAddendum": [Function],
        },
        "assign": [Function],
      },
      "cloneElement": [Function],
      "createContext": [Function],
      "createElement": [Function],
      "createFactory": [Function],
      "createRef": [Function],
      "forwardRef": [Function],
      "isValidElement": [Function],
      "lazy": [Function],
      "memo": [Function],
      "useCallback": [Function],
      "useContext": [Function],
      "useDebugValue": [Function],
      "useEffect": [Function],
      "useImperativeHandle": [Function],
      "useLayoutEffect": [Function],
      "useMemo": [Function],
      "useReducer": [Function],
      "useRef": [Function],
      "useState": [Function],
      "version": "16.11.0",
    }
  }
>
  <span
    className="flag-icon flag-icon-km"
  />
</FlagIcon>
`;
